app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsRepository.scala

```diff
@@ -19,20 +19,25 @@ package uk.gov.hmrc.apisubscriptionfields.repository
 import javax.inject.{Inject, Singleton}
 
 import com.google.inject.ImplementedBy
+import play.api.Logger
 import play.api.libs.json._
 import reactivemongo.api.indexes.IndexType
 import reactivemongo.bson.BSONObjectID
+import reactivemongo.play.json._
 import reactivemongo.play.json.collection.JSONCollection
 import uk.gov.hmrc.apisubscriptionfields.model._
 import uk.gov.hmrc.mongo.ReactiveRepository
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
+
 @ImplementedBy(classOf[SubscriptionFieldsMongoRepository])
 trait SubscriptionFieldsRepository {
+  type IsInsert = Boolean
```

Comment:
Why did you create this alias? 
What is the purpose of it?

Reply:
It it to make the code more readable by revealing the intention of the type. eg below we can more easily see why `!updateLastError.updatedExisting` is being inverted
``` scala
    mongoCollection.findAndModify(selector, updateOp).map {findAndModifyResult =>
      val maybeTuple: Option[(SubscriptionFields, IsInsert)] = for {
        value <- findAndModifyResult.value
        updateLastError <- findAndModifyResult.lastError
      } yield (value.as[SubscriptionFields], !updateLastError.updatedExisting)
```

Reply:
Ok.
It makes sense. The code is more readable.
I would ask you two things:
1) Could you please move this alias to the model?
2) Could you replace `Boolean` with this alias in the relevant code for all layers (repository, service, controller) for both field definitions and subscription fields?

Reply:
I have pushed extended scope of this improvement